tools/Repo-Tasks.psm1

```diff
@@ -0,0 +1,245 @@
+{
+
+    if([string]::IsNullOrEmpty($uid) -eq $false)
+    {
+        Write-Host "UserId=" -ForegroundColor Green -NoNewline
+        Write-Host $uid";" -NoNewline 
+    }
+
+    if([string]::IsNullOrEmpty($pwd) -eq $false)
+    {
+        Write-Host "Password=" -ForegroundColor Green -NoNewline
+        Write-Host $pwd";" -NoNewline 
+    }
+
+    if([string]::IsNullOrEmpty($subId) -eq $false)
+    {
+        Write-Host "SubscriptionId=" -ForegroundColor Green -NoNewline
+        Write-Host $subId";" -NoNewline 
+    }
+
+    if([string]::IsNullOrEmpty($aadTenant) -eq $false)
+    {
+        Write-Host "AADTenant=" -ForegroundColor Green -NoNewline
+        Write-Host $aadTenant";" -NoNewline
+    }
+
+    if([string]::IsNullOrEmpty($spn) -eq $false)
+    {
+        Write-Host "ServicePrincipal=" -ForegroundColor Green -NoNewline
+        Write-Host $spn";" -NoNewline
+    }
+
+    if([string]::IsNullOrEmpty($spnSecret) -eq $false)
+    {
+        Write-Host "ServicePrincipalSecret=" -ForegroundColor Green -NoNewline
+        Write-Host $spnSecret";" -NoNewline
+    }
+
+    if([string]::IsNullOrEmpty($recordMode) -eq $false)
+    {
+        Write-Host "HttpRecorderMode=" -ForegroundColor Green -NoNewline
+        Write-Host $recordMode";" -NoNewline
+    }
+
+    if([string]::IsNullOrEmpty($targetEnvironment) -eq $false)
+    {
+        Write-Host "Environment=" -ForegroundColor Green -NoNewline
+        Write-Host $targetEnvironment";" -NoNewline
+    }
+
+    if([string]::IsNullOrEmpty($uris) -eq $false)
+    {
+        Write-Host "BaseUri=" -ForegroundColor Green -NoNewline
+        Write-Host $uris";" -NoNewline
+    }
+
+    Write-Host ""
+}
+
+[CmdletBinding]
+Function Get-BuildScopes
+{
+<#
+.SYNOPSIS
+You can build a particular package rather than doing a full build by providing Build Scope.
+This cmdlet will help to identify existing Scope available
+This will enable to execute Start-RepoBuild <scope>
+
+#>
+
+    Write-Host "Below are available scopes you can specify for building specific projects"
+    Write-Host ""    
+    Get-ChildItem -path "$env:repoRoot\src\ResourceManager" -dir | Format-Wide -Column 5 | Format-Table -Property Name    
+    Get-ChildItem -path "$env:repoRoot\src\ServiceManagement" -dir | Format-Wide -Column 5 | Format-Table -Property Name
+}
+
+[CmdletBinding]
+Function Start-Build
+{
+<#
+.SYNOPSIS
+This cmdlet will help to do either with full build or targeted build for specific scopes.
+
+.PARAMETER BuildScope
+Use Get-BuildScope cmdLet to get list of existing scopes that can be used to build
+#>
+    param(
+    [parameter(Mandatory=$false, Position=0, HelpMessage='BuildScope that you would like to use. For list of build scopes, run List-BuildScopes')]
+    [string]$BuildScope
+    )    
+    
+    if([string]::IsNullOrEmpty($BuildScope) -eq $true)
+    {
+       Write-Host "Starting Full build"
+       msbuild.exe "$env:repoRoot\build.proj" /t:Build
```

Comment:
Same comment:  We need to make sure we are using the correct version of MSBuild, as in here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1243/ We should expand this script to include later versions of VS, and ensure the correct environment is used